### PR TITLE
rgw: solve warning: control reaches end of non-void function

### PR DESCRIPTION
**Pull request by iotcg**
src/rgw/rgw_rest_s3.cc: In member function ‘RGWOp* RGWHandler_REST_Bucket_S3::get_obj_op(bool)’:
src/rgw/rgw_rest_s3.cc:3588:7: warning: control reaches end of non-void function [-Wreturn-type]

Fixes: 4ffc765c4c5d (rgw:listobjectsv2)

Signed-off-by: Changcheng Liu <changcheng.liu@aliyun.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by iotcg** [comment]
@albIN7 Please help check

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -3581,7 +3581,7 @@ RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
   if (get_data) {   
     if (list_type == 1) {
        return new RGWListBucket_ObjStore_S3;     
-    } else if(list_type == 2) {
+    } else { /* list_type: 2 */
       return new RGWListBucket_ObjStore_S3v2;
     } } else {
```
Comment (albIN7) [inline src/rgw/rgw_rest_s3.cc]: Oops I missed the warning. Thanks for pointing that @iotcg . As of now, we are making listobjectsv1 as the default. So, this won't work. A better approach is:

  if (get_data) {   
    if (list_type == 2) {
       return new RGWListBucket_ObjStore_S3v2;     
    } else {
      return new RGWListBucket_ObjStore_S3;
    }
}
     else {
    return new RGWStatBucket_ObjStore_S3;    
  }

Comment (albIN7) [reply src/rgw/rgw_rest_s3.cc]: There is a similar pr: https://github.com/ceph/ceph/pull/28809
Could you please close this? Really appreciate your effort @iotcg

Comment (iotcg) [reply src/rgw/rgw_rest_s3.cc]: Thanks for your remind. Close now.

**Comment by iotcg** [comment]
Fixed in : https://github.com/ceph/ceph/pull/28809

**Pull request closed without merging**